cassovary-core/src/main/scala/com/twitter/cassovary/graph/MemoryMappedDirectedGraph.scala

```diff
@@ -0,0 +1,153 @@
+package com.twitter.cassovary.graph
+
+import java.io.{BufferedOutputStream, DataOutputStream, File, FileOutputStream}
+
+import com.twitter.cassovary.graph.StoredGraphDir.StoredGraphDir
+import com.twitter.cassovary.util.NodeNumberer
+import com.twitter.cassovary.util.io.{AdjacencyListGraphReader, IntLongSource, MemoryMappedIntLongSource}
+
+/**
+ * A graph which reads edge data from a memory mapped file.  There is no object overhead per node: the memory
+ * used for n nodes and m edges with both in-neighbor and out-neighbor access is exactly 8 + 16*n + 8*m bytes.
+ * Also, Loading is very fast because no parsing of text is required.  Loading time is exactly the time it takes the operating system
+ * to map data from disk into memory.
+ *
+ * Nodes are numbered sequentially from 0 to nodeCount - 1 and must be a range of this form (i.e. nodeCount == maxNodeId + 1).
+ * When transforming a graph where nodeCount <= maxNodeId to this format, new nodes with no neighbors will be implicitly created.
+ *
+ * Currently only supports storing both in-neighbors and out-neighbors of nodes (StoredGraphDir.BothInOut).
+ * The binary format is currently subject to change.
+ */
+
+/* Storage format
+byteCount  data
+4          (reserved, later use for versioning or indicating undirected vs directed)
```

Comment:
OK, it's fine to use 8 bytes.  To fully support 8 byte nodeIds we would need to change the size of the neighbor arrays to 8 bytes per node, and probably add some compression since most of the high order bits will be 0.